### ISSUE TITLE
[0.4.3] restrict-plus-operands misses aliased any and any-typed interface properties

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1263]
## Summary

With explicit allowAny:false, the rule catches a directly annotated any operand but misses an equivalent any type alias or an interface property annotated any. Number aliases and properties provide valid controls.

Verified with @utoo/lint 0.4.3, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. This reproduction does not depend on different allowAny defaults between plugin versions.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.3 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each snippet separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## Reproductions

```tsx
type Value=any;export function example(a:Value){return a+1;}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
(no diagnostics)
```

```tsx
interface Data{value:any;}export function example(a:Data){return a.value+1;}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
(no diagnostics)
```

## Controls

```tsx
export function example(a:any){return a+1;}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `any`.
```

```tsx
type Value=number;export function example(a:Value){return a+1;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

```tsx
interface Data{value:number;}export function example(a:Data){return a.value+1;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

There are no parsing failures or lint-disable comments, and only this rule is enabled.
